tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -160,32 +167,41 @@ def histograms_impl(self, tag, run, downsample_to=None):
       # can be formally expressed as the following:
       #   [s_min + math.ceil(i / k * (s_max - s_min)) for i in range(0, k + 1)]
       cursor.execute(
-          '''
+        '''
+        SELECT
+          MIN(step) AS step,
+          computed_time,
+          data,
+          dtype,
+          shape
+        FROM Tensors
+        INNER JOIN (
           SELECT
-            MIN(step) AS step,
-            computed_time,
-            data,
-            dtype,
-            shape
+            MIN(step) AS min_step,
+            MAX(step) AS max_step
           FROM Tensors
-          INNER JOIN (
-            SELECT
-              MIN(step) AS min_step,
-              MAX(step) AS max_step
-            FROM Tensors
-            /* Filter out NULL so we can use TensorSeriesStepIndex. */
-            WHERE series = :tag_id AND step IS NOT NULL
-          )
-          /* Ensure we omit reserved rows, which have NULL step values. */
+          /* Filter out NULL so we can use TensorSeriesStepIndex. */
           WHERE series = :tag_id AND step IS NOT NULL
-          /* Bucket rows into sample_size linearly spaced buckets, or do
-             no sampling if sample_size is NULL. */
-          GROUP BY
-            IFNULL(:sample_size - 1, max_step - min_step)
-            * (step - min_step) / (max_step - min_step)
-          ORDER BY step
-          ''',
-          {'tag_id': tag_id, 'sample_size': downsample_to})
+        )
+        JOIN Tags
+          ON Tensors.series = Tags.tag_id
+        JOIN Runs
+          ON Tags.run_id = Runs.run_id
+        /* Ensure we omit reserved rows, which have NULL step values. */
+        WHERE 
+          /* For backwards compatibility, ignore the experiment id
```

Comment:
It doesn't look like this PR as written actually uses the `experiment` parameter for anything - the `:exp` SQL parameter isn't actually passed any value.  In that case, can we leave the SQL code unchanged?